applications/StructuralMechanicsApplication/tests/test_truss_adjoint_element_3d2n.py

```diff
@@ -0,0 +1,288 @@
+        index = index + 1
+        disp[index] = node.GetSolutionStepValue(KratosMultiphysics.DISPLACEMENT_Y,0)
+        index = index + 1
+        disp[index] = node.GetSolutionStepValue(KratosMultiphysics.DISPLACEMENT_Z,0)
+        index = index + 1
+ 
+def add_variables(mp):
+    mp.AddNodalSolutionStepVariable(KratosMultiphysics.DISPLACEMENT)
+    mp.AddNodalSolutionStepVariable(KratosMultiphysics.REACTION)
+    mp.AddNodalSolutionStepVariable(KratosMultiphysics.VOLUME_ACCELERATION)
+    mp.AddNodalSolutionStepVariable(KratosMultiphysics.VELOCITY)
+    mp.AddNodalSolutionStepVariable(KratosMultiphysics.ACCELERATION)
+
+def apply_material_properties(mp,dim):
+    #define properties
+    mp.GetProperties()[1].SetValue(KratosMultiphysics.YOUNG_MODULUS,1000)
+    mp.GetProperties()[1].SetValue(KratosMultiphysics.DENSITY,7850)
+    mp.GetProperties()[1].SetValue(StructuralMechanicsApplication.CROSS_AREA,10)
+    mp.GetProperties()[1].SetValue(StructuralMechanicsApplication.TRUSS_PRESTRESS_PK2,5)
+    g = [0,0,0]
+    mp.GetProperties()[1].SetValue(KratosMultiphysics.VOLUME_ACCELERATION,g)
+    cl = StructuralMechanicsApplication.TrussConstitutiveLaw()
+    mp.GetProperties()[1].SetValue(KratosMultiphysics.CONSTITUTIVE_LAW,cl)
+
+def copy_solution_step_data_of_node(node, mp_old, node_id, step=0):
+    node.SetSolutionStepValue(KratosMultiphysics.DISPLACEMENT_X,step,
+    mp_old.Nodes[node_id].GetSolutionStepValue(KratosMultiphysics.DISPLACEMENT_X,0))
+    node.SetSolutionStepValue(KratosMultiphysics.DISPLACEMENT_Y,
+    mp_old.Nodes[node_id].GetSolutionStepValue(KratosMultiphysics.DISPLACEMENT_Y,0))
+    node.SetSolutionStepValue(KratosMultiphysics.DISPLACEMENT_Z,
+    mp_old.Nodes[node_id].GetSolutionStepValue(KratosMultiphysics.DISPLACEMENT_Z,0))
+
+def shape_perturbation_correction_factor(node_1, node_2):
+    dx = node_1.X - node_2.X
+    dy = node_1.Y - node_2.Y
+    dz = node_1.Z - node_2.Z
+    l = math.sqrt(dx*dx + dy*dy + dz*dz)
+    return l    
+  
+def zero_vector(size):
+    v = KratosMultiphysics.Vector(size)
+    for i in range(size):
+        v[i] = 0.0
+    return v
+
+def create_property_perturbed_elements(mp, delta, new_element_name):
+    dim = 3
+    model_part = KratosMultiphysics.ModelPart("Property_Perturbed_Elements")
+    model_part.ProcessInfo.SetValue(KratosMultiphysics.DOMAIN_SIZE,dim)
+    add_variables(model_part)
+    model_part.CreateNewNode(1, mp.Nodes[1].X, mp.Nodes[1].Y, mp.Nodes[1].Z)
+    model_part.CreateNewNode(2, mp.Nodes[2].X, mp.Nodes[2].Y, mp.Nodes[2].Z)
+    apply_material_properties(model_part,dim)
+
+    A_initial = mp.GetProperties()[1][StructuralMechanicsApplication.CROSS_AREA]
+    model_part.GetProperties()[1].SetValue(StructuralMechanicsApplication.CROSS_AREA, A_initial + delta )
+    prop = model_part.GetProperties()[1]
+
+    model_part.CreateNewElement(new_element_name, 1, [1, 2], prop)
+
+    for i in range(2):
```

Comment:
`for node, old_node in zip(model_part, old_model_part):`

Reply:
done